### PR TITLE
Abstract document add module info file

### PR DESCRIPTION
**Pull request by saksham93**
Pull request title

Abstract document add module info file. Refers to #1047 

Pull request description

Added module-info.java file in Abstract Document Design pattern.

If this is the correct way, will add it to more design patterns.

### CODE REVIEW

**Comment by saksham93** [comment]
@iluwatar Can you please suggest if this PR is ok? Thank you.

File: abstract-document/src/main/java/com/iluwatar/abstractdocument/module-info.java [hunk 1]
```diff
@@ -0,0 +1,2 @@
+module abstractdocument.modules {
+}
```
Comment (iluwatar) [inline abstract-document/src/main/java/com/iluwatar/abstractdocument/module-info.java]: I would prefer format like this:
```
module com.iluwatar.abstractdocument {
}

```

Comment (iluwatar) [reply abstract-document/src/main/java/com/iluwatar/abstractdocument/module-info.java]: The build is failing with `package org.slf4j is not visible
  (package org.slf4j is declared in the unnamed module, but module org.slf4j does not read it)`

To fix it you need to declare something like this inside the module-info:
`requires org.slf4j;`

Comment (saksham93) [reply abstract-document/src/main/java/com/iluwatar/abstractdocument/module-info.java]: @iluwatar Thank you for your comments. I have done the required changes. Can you please suggest?

**Comment by iluwatar** [comment]
https://github.com/iluwatar/java-design-patterns/issues/1047

**Comment by iluwatar** [comment]
Thanks @saksham93, first one down only a few to go! :relaxed:

**Pull request merged**